src/Security/Authentication/test/AuthenticationMiddlewareTests.cs

```diff
@@ -48,6 +48,68 @@ public async Task OnlyInvokesCanHandleRequestHandlers()
             Assert.Equal(607, (int)response.StatusCode);
         }
 
+        [Fact]
+        public async Task CanUseAuthenticationResultFeature()
+        {
+            var ticket = new AuthenticationTicket(new ClaimsPrincipal(), "Default");
+            var result = AuthenticateResult.Success(ticket);
+            SimpleHandler.Result = result;
+
+            using var host = new HostBuilder()
+                .ConfigureWebHost(webBuilder =>
+                {
+                    webBuilder.ConfigureServices(services => services.AddAuthentication(o =>
+                    {
+                        o.AddScheme("Default", s =>
+                        {
+                            s.HandlerType = typeof(SimpleHandler);
+                        });
+                        o.DefaultScheme = "Default";
+                    }))
+                    .Configure(app => 
+                    { 
+                        app.UseAuthentication();
+                        app.Use((context, next) =>
+                        {
+                            var auth = context.Features.Get<IAuthenticationResultFeature>();
```

Comment:
There should be an extension method added to retrieve the new data. Users shouldn't need to directly access features for common operations.

Reply:
So is this really better than just having them call context.AuthenticateAsync(), that usually is equivalent to this.  But we could add a sugar method that uses this feature called context.GetDefaultAuthenticateResult() I guess which is just a cache on top of our cached result that you'd normally get back from AuthenticateAsync...

Reply:
Yeah, the value add here is pretty suspect. @davidfowl

Reply:
I should share why I wanted this. In SignalR the HttpContext sometimes goes away (long polling) but we want to get the user and authentication result so we can do expiry checks and abort the connection if the auth ticket expires.

Reply:
Okay so for that situation, its mostly about getting access to the default authentication the middleware did, since you can't even call Authenticate again if you wanted...

Does it seem reasonable to only target the no HttpContext case for this feature then?  Since if you have httpContext, just call AuthenticateAsync() to get the default result.  Maybe we call this DefaultAuthenticateResultFeature, and then it maps exactly to capturing the AuthMiddleware's result and nothing more.  It doesn't need to have any interaction with HttpContext at all?

Reply:
I'd expect you'd still want to set it in the MVC and SignalR auth filters, those update the user based on the current authz policy don't they?

Reply:
Yeah but that doesn't really work, you can combine a User out of the ClaimsPrincipals, but there's a reason we don't preserve the meta data there, since which one would you put in there.

Reply:
Then this preserved result data is misleading if it's not going to match the HttpContext.User in MVC or SignalR.

Alternatively, make this a `List<AuthenticationResult>` so you can preserve all of them.

https://github.com/dotnet/aspnetcore/blob/b879e587370fc669a3433e89c5311423e6bb1ee1/src/Security/Authorization/Policy/src/PolicyEvaluator.cs#L40-L44